File: 733-flood-fill/flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        xMod = [1,0,-1,0]
        yMod = [0,-1,0,1]

        n = len(image)
        m = len(image[0])

        origColor = image[sr][sc]
        newColor = color

        def dfs(x,y,origColor,newColor):

            if ( not valid(x,y)):
                return
            if ( image[x][y] == newColor):
                return
            
            if ( image[x][y] == origColor):
                image[x][y] = newColor

                for i in range(0,4):
                    dfs(x + xMod[i], y + yMod[i],origColor,newColor)

        def valid(x,y):
            if x < 0 or x >= n:
                return False
            if y < 0 or y >= m:
                return False
            return True

        dfs(sr,sc,image[sr][sc],color)
        return image

File: 733-flood-fill/flood-fill.cpp
class Solution {
public:
    int m;
    int n ;

    void DFS(int i, int j ,vector<vector<int>> &image,int oldColor , int newColor )
    {
        if ( i < 0 || i >= m) return ;
        if ( j < 0 || j >= n) return ;
        if ( image[i][j] != oldColor) return ;

        if ( image[i][j] == oldColor)
        {
            image[i][j] = newColor;
            DFS(i+1,j,image,oldColor,newColor);
            DFS(i,j-1,image,oldColor,newColor);
            DFS(i-1,j,image,oldColor,newColor);
            DFS(i,j+1,image,oldColor,newColor);


        }
        

    }
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) 
    {
        m = image.size();
        n = image[0].size();
        int oldColor = image[sr][sc];

        if (oldColor == color) return image;
        DFS(sr,sc,image,oldColor,color);

        return image;

        
    }
};